### DISCUSSION TITLE
CSV Import Unavailable

### DISCUSSION THREAD

**Discussion by washroomapricot9** [firefly-iii/firefly-iii#6016] [Firefly III]
Hi All,

First time user of Firefly here. I managed to get the core application running today using mariadb, both in docker on my Synology NAS. I've been trying to import some data to have an 'initial play' using the FIDI docker package but for some reason CSV import isn't available to me as it has a spinning gear/cog (see screenshot below) and doesn't react when I click it. Any ideas why this is? The server seems to be authenticated fine and neither of the red buttons fix it. 

![Screenshot 2022-04-19 215303](https://user-images.githubusercontent.com/102882928/164093763-070d7466-e86a-4cb2-8e61-2d0159804b78.jpg)

Thanks in advance

**Comment by JC5** [comment]
I think it's not capable of reaching Firefly III. Perhaps it's not configured to the right (internal) IP address?

**Reply by washroomapricot9** [reply]
Hi, thanks for getting back to me. I'm pretty sure it's configured to the correct address (tried both http://[INTERNAL-IP]:[PORT NUMBER] & http://[HOSTNAME]:[PORT NUMBER]). I've tried removing the non standard environment settings from docker and inputting the data through the webpage instead which looked more promising at first but then failed. I used a (non confidential) Client ID generated under the OAuth Clients section in Firefly along with the addresses detailed above which led me to the following (which has the address of the Firefly instance in the browser bar):

![image](https://user-images.githubusercontent.com/102882928/165340926-4db20752-9441-4079-be70-171bba9b7e89.png)

However, after clicking 'Authorise', nothing happens for 30 seconds or so and then I'm presented with this (which has the address of the FIDI instance in the browser bar):

![image](https://user-images.githubusercontent.com/102882928/165340827-802a888e-4e23-42f3-aa68-bbd89eb7aec5.png)

Should also note, I'm running the :latest tags of both docker packages, which were downloaded 7 days ago. Firefly: 5.7.2, FIDI: 0.9.8

**Reply by JC5** [reply]
Thanks for replying!

I expect the 500 error can be traced back to the FIDI logs. `docker logs -f [fidi container]`. Let me know what it says, a stack trace should allow me to trace the error. Thanks!

**Reply by washroomapricot9** [reply]
I've just sussed it out, it was an issue with port 53 (DNS) and the Firefly port being blocked on the network 172.17.xxx.xxx internal Docker network. Thanks for your help anyway, keep up the good work! Time to have a play 👍

**Reply by JC5** [reply]
You're welcome, enjoy!
